Add workflow document (#533)
iotaledger/identity.rs: documentation/docs/workflow.md

### Work Branches

These are branches that developers work on directly. Their names should be prefixed appropriately with one of the following categories. For example, a PR fixing a null pointer bug in the Wasm bindings might be created from a branch called `fix/client-non-null`.

#### Feature (feat/, doc/, chore/, fix/)
Singular PR contributions should create either a `feat`, `doc`, `chore`, or `fix` branch, depending on the type of changes. These may be branched from either the `dev` branch or an `epic` branch. If the number of lines of code is going to be relatively small and the work completed in a single PR, the branch should be created from `dev` and merged back into `dev` once completed. Otherwise, the branches should be created from their associated `epic` branch and be merged back into the same `epic` branch. 

- `feat` branches should contain changes to the code that expand or modify functionality. They should also include updates to the documentation and examples related to the feature, though `doc` branches may be used to catch up on documenting a feature.

> - `doc` branches contain changes to code documentation or the wiki. These PRs should be kept relatively small in order to not burden a reviewer with too many documentation updates. For example, during a documentation catch-up, we will have a branch/PR per documentation page.

Reviewer: ```suggestion
- `doc` branches contain changes to code documentation or the wiki. These PRs should be kept relatively small in order to avoid burdening a reviewer with too many documentation updates at once. For example, during a documentation catch-up we will have a branch/PR per documentation page.
```

- `chore` branches are short lived branches that contain no significant features or functionality changes, but rather smaller fixes such as typos, code fixes, minor refactors, and CI changes.
- `fix` branches correct bugs such as compilation errors or where existing features do not behave as expected, generally without introducing any new functionality or breaking changes.

> It is recommended to integrate `dev` or `epic` regularly, depending on from where the branch started, to reduce the possibility and potential size of merge conflicts.

Reviewer: ```suggestion
It is recommended to integrate `dev` or `epic` regularly, depending on from where the branch started, to reduce the possibility and size of merge conflicts.
```

Author: @cycraig If changes are merged into the epic branch (by other developers) or `dev` is merged into the epic branch IMHO it makes sense to integrate them.

#### Epic (epic/)
Long-lived `epic` branches should be created as soon as a feature is expected to require more than one PR. The `epic` branch should be branched from `dev` and should only accept merges that are related to the feature being developed. A PR should be opened as soon as the branch is created to publicly notify contributors about the development, the goals and requirements of the feature and the existence of the branch. It is recommended to integrate `dev` often to reduce the possibility and potential size of merge conflicts.

### Semantic Versioning

> Semantic Versioning (SemVer) describes a methodology for versioning of software to convey meaning and guarantees through the version string. A typical version string looks like `2.3.1`, where `2` is called the major, `3` the minor and `1` the patch or bugfix version. 

Reviewer: ```suggestion
Semantic Versioning (SemVer) describes a methodology for versioning of software to convey meaning and guarantees through the version string. A typical version string looks like `2.3.1`, where `2` is called the major version, `3` the minor version and `1` the patch or bugfix version. 
```

The central idea is that every part of the version string conveys meaning. A major change introduces behavior that is incompatible with previous versions of the software, while a minor change adds backwards-compatible functionality and a patch simply fixes a problem. So just by looking at the version string an implementer will have an understanding of the effort needed to integrate a new version.
For more detailed information and an overview of advanced features see [Semantic Versioning 2.0.0](https://semver.org/). Not to be confused with [Sentimental Versioning](http://sentimentalversioning.org/).

### Changelog

> A changelog is a file describing a software project for humans to grasp the type and content of changes from version to version. Changelogs are closely related to the versioning of software, since individual changes are grouped into versions that are in our case referenced by a [SemVer string](#Semantic-Versioning). We generally follow the recommendations from [keepachangelog](https://keepachangelog.com/en/1.0.0/). The changelog in this project is generated from the titles and [labels](#PR-Labels) of [Pull-Requests](#Pull-Requests). 

Reviewer: ```suggestion
A changelog is a file describing a software project for humans to grasp the type and content of changes from version to version. Changelogs are closely related to the versioning of software, since individual changes are grouped into versions that are in our case referenced by a [SemVer string](#Semantic-Versioning). We generally follow the recommendations from [keepachangelog](https://keepachangelog.com/en/1.0.0/). The changelog in this project is generated from the titles and [labels](#PR-Labels) of [pull requests](#Pull-Requests). 
```

#### PR Labels

> Labels are used to categorize changes in [Pull-Requests](#Pull-Requests). Adding a label will include the labeled [Pull-Request](#Pull-Requests) in the related section of the generated [Changelog](#Changelog).

Reviewer: ```suggestion
Labels are used to categorize changes in [pull requests](#Pull-Requests). Adding a label will include the labeled [PR](#Pull-Requests) title in the related section of the generated [changelog](#Changelog).
```

##### `Rust` 
Includes the PR in the core Rust library changelog.

##### `Wasm`
Includes the PR in the Wasm bindings changelog.

It is also necessary to add an appropriate label for the type of change in the PR. The following labels determine in which section a PR title will appear: 

##### Changed
Maps to the major version of [Semantic Versioning](#Semantic-Versioning).
labels: `Breaking change`

##### Added
Maps to the minor version of [Semantic Versioning](#Semantic-Versioning).
labels: `Added`

##### Patch
Maps to the patch version of [Semantic Versioning](#Semantic-Versioning).
labels: `Patch`

##### Deprecated
Marks features that will be removed in the feature. No special version consideration apply here, since the feature did not change yet.
labels: `Deprecated`

##### Removed
Marks features as being removed. Typically the features should have been deprecated in the previous version. This maps to the major version of [Semantic Versioning](#Semantic-Versioning). 
labels: `Removed`

##### Excluded Tags
Marks changes that should not be part of the changelog. This should only be used for documentation and rare exceptions
labels: `Documentation`, `No changelog`

> Please note that a PR can only be listed in one section of a changelog. So attaching the labels `Rust` `Added` `Patch` to a PR is invalid.

Reviewer: ```suggestion
Please note that a PR can only be listed in one section of a changelog. So attaching the labels `Rust` `Added` `Patch` to a PR, for example, is invalid because `Added` and `Patch` conflict.
```

##### Release Summary
To attach a release summary to a version in the changelog, an issue with the label `release-summary` must be created. Create a GitHub milestone matching the version you want to describe and attach it to the issue. The issue can be closed immediately. The text of the issue will be included in the changelog as the release summary.

### Issue Labels
The following labels are used to categorize issues but do not have any effect on changelogs: `Request`, `Enhancement`, `Bug`, `Chore`, `Dependencies`, `Help wanted`, `Duplicate`, `Wontfix`.

## Release

With the release process we can deliver versions of our software to the community. We use sensible automation where it helps to remove tedium. However, some steps that require active decision making remain manual.

> The final list of changes from the [changelog](#Changelog) informs the version of the release. If at least one change mapping to a major version is included, the major version needs to be incremented. In that case the minor and patch version are set to `0`. If there are no change related to a major version, but changes related to a minor version are present the minor version needs to be incremented while the patch version is set to `0`. Otherwise only the patch version is incremented. Determining the version of the release is the responsibility of the person performing the release.
> The determined version of the release is used to create the [hotfix](#hotfix) or [release](#release) branch. For example a major release from the previous version `v2.3.1` will create the `release/v3.0.0` branch.
> Notice the `v` in front of the version. We [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) all release in git in the form of `vMAJOR.MINOR.PATCH`. For bindings we prefix the tag with the binding name, so a tag for Wasm would look like `wasm-v1.2.3`. Bindings and the core Rust library are versioned and released independently.

Reviewer: ```suggestion
The final list of changes from the [changelog](#Changelog) informs the version of the release. If at least one change mapping to a major version is included, the major version needs to be incremented. In that case the minor and patch versions are set to `0`. If there are no change related to a major version, but changes related to a minor version are present, the minor version needs to be incremented while the patch version is set to `0`. Otherwise only the patch version is incremented. Determining the version of the release is the responsibility of the person performing the release.

The determined version of the release is used to create the [hotfix](#hotfix) or [release](#release) branch. For example, a major release from the previous version `v2.3.1` will create the `release/v3.0.0` branch.
Notice the `v` in front of the version. We [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) all release in git in the form of `vMAJOR.MINOR.PATCH`. For bindings we prefix the tag with the binding name, so a tag for Wasm would look like `wasm-v1.2.3`. Bindings and the core Rust library are versioned and released independently.
```

Other contributors are encouraged to provide feedback on a PR during its development. A PR should be flagged as 'ready for review' once the PR has implemented all changes and no further commits are planned by the main contributors.
The repository requires a review to be provided by at least one (other) developer in the team that works in the same language or has knowledge of the work before it can be merged.
In order to generate good [changelogs](#Changelog), the PR title must be written in a way that is suitable as an changelog entry and the PR needs to be [labeled](#PR-Labels) correctly.
Once a PR is approved, the preferred method is "squash-and-merge" to keep the destination branch clean and allow many small commits while work is in-progress.

Linting: `cargo clippy --all-targets --all-features -- -D warnings`

> Formatting: `cargo fmt --all`

Author: Suggestion from @JelleMillenaar is to remove this here and maybe extend it into a separate document later.

### Work Branches

These are the branches that developers work on, and their names should be prefixed with the appropriate category, described in the following. For example, a PR fixing a null pointer bug in the Wasm bindings might be created from a branch called `fix/client-non-null`.

#### Feature (feat/, doc/, chore/, fix/)
Singular PR contributions should create either `feat`, `doc` or `chore` branches, depending on the type of work to be done. `feat`, `doc` and `chore` branches may be branched from either the `dev` branch or an `epic` branch. If the number of lines of code is going to be relatively small and the work will be completed in a single PR, the branch should be created from `dev` and merged back into `dev` once completed. Otherwise, the branches should be created from their associated `epic` branch and be merged back into the same `epic` branch. 

> Any `feat` branch should include updates to the documentation and examples related to the feature. 
> It might be necessary to create a dedicated `doc` PR to, for example, catch up on documenting a feature. These `doc` PRs should be kept relatively small in order to not burden a reviewer with too many documentation updates. For example, during the documentation catch-up, we will have a branch/PR per documentation page.
> `chore` branches are short lived branches that contain no significant features or code updates, but rather smaller fixes such as typo's, code fixes and CI changes.

Reviewer: Formatted as a list and added the missing `fix` branch.

```suggestion
- `feat` branches should contain changes to the code that expand or modify functionality. They should also include updates to the documentation and examples related to the feature, though `doc` branches may be used to catch up on documenting a feature.
- `doc` branches contain changes to code documentation or the wiki. These PRs should be kept relatively small in order to not burden a reviewer with too many documentation updates. For example, during a documentation catch-up, we will have a branch/PR per documentation page.
- `chore` branches are short lived branches that contain no significant features or functionality changes, but rather smaller fixes such as typos, code fixes, minor refactors, and CI changes.
- `fix` branches correct bugs such as compilation errors or where existing features do not behave as expected, generally without introducing any new functionality or breaking changes.
```

It is recommended to integrate `dev` or `epic` regularly, depending on from where the branch started, to reduce the possibility and potential size of merge conflicts.

> #### Epic (epic/)
> Long-lived `epic` branches should be created and pushed to `origin` as soon as a feature is expected to require more than one PR. The `epic` branch should be branched from `dev` and should only accept merges that are related to the feature being developed. A PR should be opened as soon as the branch is created to publicly notify contributors about the development, the goals and requirements of the feature and the existence of the branch. It is recommended to integrate `dev` often to reduce the possibility and potential size of merge conflicts.

Reviewer: ```suggestion
#### Epic (epic/)
Long-lived `epic` branches should be created as soon as a feature is expected to require more than one PR. The `epic` branch should be branched from `dev` and should only accept merges that are related to the feature being developed. A PR should be opened as soon as the branch is created to publicly notify contributors about the development, the goals and requirements of the feature and the existence of the branch. It is recommended to integrate `dev` often to reduce the possibility and potential size of merge conflicts.
```

> #### Release (release/) {#release}
> Release branches allow to move changes from `dev` to `main`. They must be created from `dev`. If version strings need to be updated in files (like `package.json` e.g.) this should happen in commits on this branch. These branches need to be merged to `main` and `dev`. Since release branches will lead to a [release](#Release) they must contain a [changelog](#Changelog) and must be versioned.

Reviewer: It's not strictly true that release branches move changes from `dev` to `main` any more since we also use them for dev releases?

I think this section needs to mention that the new GitHub Actions workflows are used create release branches automatically.

```suggestion
#### Release (release/) {#release}
Release branches move changes from `dev` to `main`. If version strings need to be updated in files (like `package.json` e.g.) this should happen in commits on this branch. These branches need to be merged to `main` and `dev`. Since release branches will lead to a [release](#Release) they must contain a [changelog](#Changelog) and must be versioned.

A `release` branch and [changelog](#Changelog) are usually created automatically using GitHub Actions. See the [release](#Release) section for more information.
```

#### Hotfix (hotfix/) {#hotfix}

> Hotfix branches allow to fix critical bugs on the `main` branch, without the need to merge them in the `dev` branch first. They must be created from `main`. If version strings need to be updated in files (like `package.json` e.g.) this should happen in commits on this branch. These branches need to be merged to `main` and `dev`, so that both branches contain the fix. Since hotfix branches will lead to a [release](#Release) they must contain a [changelog](#Changelog) and must be versioned.

Reviewer: I think hotfix branches are going to be replaced by support branches? If not, we still need to mention support branches somewhere.

> ### Semantic Versioning
> Semantic Versioning, or SemVer, describes a methodology for versioning of software to convey meaning and guarantees through the version string. A typical version string looks like `2.3.1`, where `2` is called a major, `3` a minor and `1` a patch or bugfix version. 

Reviewer: ```suggestion
### Semantic Versioning
Semantic Versioning (SemVer) describes a methodology for versioning of software to convey meaning and guarantees through the version string. A typical version string looks like `2.3.1`, where `2` is called the major, `3` the minor and `1` the patch or bugfix version. 
```

> The central idea is, that every part in the version string conveys meaning. A major change will introduce behavior that is incompatible with previous versions of the software, a minor change adds functionality and a patch simply fixes a problem. So just by looking at the version string an implementer will have an understanding of the effort he needs to put in to integrate the new version.

Reviewer: ```suggestion
The central idea is that every part of the version string conveys meaning. A major change introduces behavior that is incompatible with previous versions of the software, while a minor change adds backwards-compatible functionality and a patch simply fixes a problem. So just by looking at the version string an implementer will have an understanding of the effort needed to integrate a new version.
```

> For more detailed information and an overview of advanced features see [Semantic Versioning 2.0.0](https://semver.org/). Also, don't do [Sentimental Versioning](http://sentimentalversioning.org/).

Reviewer: Not sure which phrasing is funnier.

```suggestion
For more detailed information and an overview of advanced features see [Semantic Versioning 2.0.0](https://semver.org/). Not to be confused with [Sentimental Versioning](http://sentimentalversioning.org/).
```

### Gitflow
Gitflow is a well established branching model for developing and releasing tightly versioned software products. It was [proposed in 2010](https://nvie.com/posts/a-successful-git-branching-model/) and has seen wide adoption. For a good introduction to the workflow and CLI integration see the [Atlassian Guide](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) or the [Gitflow Cheatsheet](https://danielkummer.github.io/git-flow-cheatsheet/index.html).

> In recent years through a shift toward contiguously delivered applications the need for strict versioning lessened and other branching models became more popular. Since this project is a library, rather then an application it needs to have strong guarantees towards implementers that we express through [Semantic Versioning](#Semantic-Versioning). Gitflow helps us to manage our work- and release-flow and to uphold those guarantees.

Reviewer: ```suggestion
In recent years, through a shift toward continuously delivered applications, the need for strict versioning lessened and other branching models became more popular. Since this project is a library rather then an application, it needs to have strong guarantees towards implementers that we express through [Semantic Versioning](#Semantic-Versioning). Gitflow helps us manage our work- and release-flow to uphold those guarantees.
```

Gitflow is a merge-based workflow, rebasing of commits is not required and not desired.

> ### Changelog
> A changelog is a file describing a software project for humans to grasp the type and content of changes from version to version. Changelogs are closely related to the versioning of software, since individual changes are grouped into versions that are in our case referenced by a [SemVer string](#Semantic-Versioning). We generally follow the recommendations from [keepachangelog](https://keepachangelog.com/en/1.0.0/). The changelog in this project is generated from the title of and [labels](#PR-Labels) attached to [Pull-Requests](#Pull-Requests). 

Reviewer: ```suggestion
### Changelog
A changelog is a file describing a software project for humans to grasp the type and content of changes from version to version. Changelogs are closely related to the versioning of software, since individual changes are grouped into versions that are in our case referenced by a [SemVer string](#Semantic-Versioning). We generally follow the recommendations from [keepachangelog](https://keepachangelog.com/en/1.0.0/). The changelog in this project is generated from the titles and [labels](#PR-Labels) of [Pull-Requests](#Pull-Requests). 
```

#### PR Labels
Labels are used to categorize changes in [Pull-Requests](#Pull-Requests). Adding a label will include the labeled [Pull-Request](#Pull-Requests) in the related section of the generated [Changelog](#Changelog).

> Changelogs are generated for the "core" and every binding separately. To attach a PR to a specific changelog use the following label:

Reviewer: ```suggestion
Changelogs are generated for the core Rust library and each binding separately. To attach a PR to a specific changelog use the following labels:
```

> 
> ##### `Rust` 
> Includes the PR in "core" / Rust changelog
> 
> ##### `Wasm`
> Includes the PR in Wasm changelog

Reviewer: ```suggestion
##### `Rust` 
Includes the PR in the core Rust library changelog.

##### `Wasm`
Includes the PR in the Wasm bindings changelog.
```

> To list the PR title in a changelog the section to list in must be described through the following labels: 

Reviewer: ```suggestion
It is also necessary to add an appropriate label for the type of change in the PR. The following labels determine in which section a PR title will appear: 
```

##### Changed
Maps to the major version of [Semantic Versioning](#Semantic-Versioning).
labels: `Breaking change`

##### Added
Maps to the minor version of [Semantic Versioning](#Semantic-Versioning).
labels: `Added`

##### Patch
Maps to the patch version of [Semantic Versioning](#Semantic-Versioning).
labels: `Patch`

##### Deprecated
Marks features that will be removed in the feature. No special version consideration apply here, since the feature did not change yet.
labels: `Deprecated`

##### Removed
Marks features as being removed. Typically the features should have been deprecated in the previous version. This maps to the major version of [Semantic Versioning](#Semantic-Versioning). 
labels: `Removed`

##### Excluded Tags
Marks changes that should not be part of the changelog. This should only be used for documentation and rare exceptions
labels: `Documentation`, `No changelog`

Please note that a PR can only be listed in one section of a changelog. So attaching the labels `Rust` `Added` `Patch` to a PR is invalid.

> ##### Release Summary
> To attach a release summary to a version in the changelog create an issues with the label `release-summary`. Create a GitHub milestone matching the version you want to describe and attach it to the issue. The issue can be closed immediately. The text of the issue will be included in the changelog as the release summary.

Reviewer: ```suggestion
##### Release Summary
To attach a release summary to a version in the changelog, an issue with the label `release-summary` must be created. Create a GitHub milestone matching the version you want to describe and attach it to the issue. The issue can be closed immediately. The text of the issue will be included in the changelog as the release summary.
```

> ### Issue Labels
> The following labels are used to categorize issues, they don't have any effect on changelogs:`Request`, `Enhancement`, `Bug`, `Chore`, `Dependencies`, `Help wanted`, `Duplicate`, `release-summary`, `Wontfix`.

Reviewer: Remove release-summary? I think it kind of affects the changelog.

```suggestion
### Issue Labels
The following labels are used to categorize issues but do not have any effect on changelogs: `Request`, `Enhancement`, `Bug`, `Chore`, `Dependencies`, `Help wanted`, `Duplicate`, `Wontfix`.
```

## Release

> With the release process we can deliver versions of our software to the community. We use sensible automation where it helps to remove tedium. Some steps and decision remain manual, since they require active decision making.

Reviewer: ```suggestion
With the release process we can deliver versions of our software to the community. We use sensible automation where it helps to remove tedium. However, some steps that require active decision making remain manual.
```

> The final list of changes from the [changelog](#Changelog) informs the version of the release. If at least one change mapping to a major version is included, the major version needs to be incremented. In this case the minor and patch version are not incremented but set to `0`. If there are no change related to a major version, but changes related to a minor version are present the minor version needs to be incremented. The major version stays untouched, the patch version is set to `0` and so on. Determining the version of the release is the responsibility of the person doing the release.

Reviewer: This kind of implies we need to generate the changelog before choosing the version, but we need to choose a version to generate a changelog.

```suggestion
The final list of changes from the [changelog](#Changelog) informs the version of the release. If at least one change mapping to a major version is included, the major version needs to be incremented. In that case the minor and patch version are set to `0`. If there are no change related to a major version, but changes related to a minor version are present the minor version needs to be incremented while the patch version is set to `0`. Otherwise only the patch version is incremented. Determining the version of the release is the responsibility of the person performing the release.
```

The determined version of the release is used to create the [hotfix](#hotfix) or [release](#release) branch. For example a major release from the previous version `v2.3.1` will create the `release/v3.0.0` branch.

> Notice the `v` in front of the version. We [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) all release in git in the form of `vMAJOR.MINOR.PATCH`. For bindings we prefix the tag with the binding name, so a tag for Wasm would look like `wasm-v1.2.3`. Bindings and the "core" are versioned and released independently.

Reviewer: ```suggestion
Notice the `v` in front of the version. We [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) all release in git in the form of `vMAJOR.MINOR.PATCH`. For bindings we prefix the tag with the binding name, so a tag for Wasm would look like `wasm-v1.2.3`. Bindings and the core Rust library are versioned and released independently.
```

> For bindings and and the "core" we may release `dev` versions. Those versions are meant as a preview of upcoming versions. For example if the current version is 1.2.3 with the tag `v1.2.3` we may release `v1.3.0-dev` which is then superseded by the actual `1.3.0` release.

Reviewer: Don't we tag dev releases with an incrementing number at the end?

```suggestion
Additionally, we may release `dev` versions separately for both bindings and the core Rust library. These releases are meant as a preview of upcoming versions. For example, if the current version is `1.2.3` with the tag `v1.2.3`, we may release `v1.3.0-dev.1` which is then superseded by the actual `1.3.0` release.
```

> To create a release follow the following steps:

Reviewer: ```suggestion
To create a release perform the following steps:
```

> 1. Make sure the changes you want to release are on the `dev` branch

Reviewer: ```suggestion
1. Ensure all the changes you want to release are on the `dev` branch.
```

> 2. Use the appropriate action, e.g. `Rust Create Release PR`

Reviewer: ```suggestion
2. Select the appropriate GitHub Actions workflow, e.g. `Rust Create Release PR`.
```

2.1. Decide if you want to create a `dev` or `main` release.
2.2. Determine the next version string.
2.3. Run the workflow. The workflow will create a PR from `dev` targeting `dev` with release related changes.
3. Review the PR
3.1 The PR will contain an updated changelog, make sure it has all expected entries in the appropriate sections and the determined version matches the changelog according to [SemVer](#Semantic-Versioning).
3.2 The PR will contain changes to files including versions or version strings. Make sure this matches expectations.

> 4. Merge the PR

Reviewer: ```suggestion
4. Merge the PR.
```

4.1. On merge an automatic workflow is triggered that optionally creates a GitHub Release (this is determined in the workflow configuration) and builds and publishes artifacts also according to the workflow configuration.
5. If the release is a `main` release, merge the `dev` branch into the `main` branch.

> ### Troubleshooting
> #### The changelog in the release PR is missing entries, has unrelated entries, entries in the wrong section.
> Fix the [labels](#PR-Labels) on the related PRs and rerun the workflow with the same parameters. The PR will be updated with the updated changelog.
> #### The release description in the release PR is missing or wrong.
> Fix the issues description, milestone and label according to the [release summaries guide](#Release-Summary) and rerun the workflow with the same parameters. The PR will be updated with the updated changelog.
> #### Features / Code is missing from the release.
> Merge the code into the `dev` branch. Rerun the workflow with the same parameters.
> #### I want to abort the release for any reason.
> CLose the PR. You can reopen it later.

Reviewer: This might need to be expanded.

```suggestion
### Troubleshooting

#### The changelog entries have the wrong description in the release PR
Update the titles of the relevant PRs then rerun the workflow with the same parameters. The release PR will be updated with the new changelog.

#### The changelog in the release PR is missing entries, has unrelated entries, entries in the wrong section.
Fix the [labels](#PR-Labels) on the relevant PRs then rerun the workflow with the same parameters. The release PR will be updated with the new changelog.

#### The release description in the release PR is missing or wrong.
Fix the issue description, milestone and label according to the [release summaries guide](#Release-Summary) and rerun the workflow with the same parameters. The release PR will be updated with the new changelog.

#### Features or code are missing from the release.
Merge the code into the `dev` branch then rerun the workflow with the same parameters. The release PR will be updated with the changes.

#### I want to abort the release for any reason.
Close the PR. You can reopen it later. // TODO: can I just re-run the workflow? Maybe that needs an "I want to resume an aborted release" section?
```
